authentication.go

```diff
@@ -70,5 +70,27 @@ func (a *App) ValidateAuthToken(authToken string) (string, bool) {
 		return "", false
 	}
 
+	expectedAud, err := a.getExpectedAudienceValue()
+	if err != nil {
+		return "", false
+	}
+
+	if !claims.VerifyAudience(expectedAud, true) {
+		return "", false
+	}
+
 	return userID, true
 }
+
+func (a *App) getExpectedAudienceValue() (string, error) {
+	appInfo, err := a.GetApp()
+	if err != nil {
+		return "", err
+	}
+
+	if appInfo.Hosted {
+		return appInfo.ID, nil
+	} else {
+		return appInfo.AuthOrigin, nil
+	}
```

Comment:
Just an update that, with the recent rollout of the updated JWT structure in production, this check can assume that every valid JWT will have the app ID as an audience claim.

So for this release, the expected audience value can just be retrieved from the `*App` struct as `a.ID`.